services/workflows-service/src/rule-engine/core/operators/helpers.ts

```diff
@@ -0,0 +1,199 @@
+      );
+    }
+
+    if (this.dataValueSchema) {
+      this.validateSchema(this.dataValueSchema, args.dataValue, `Invalid data value`);
+    }
+  }
+
+  protected validateSchema(schema: ZodSchema<any>, value: unknown, message: string) {
+    const result = schema.safeParse(value);
+
+    if (!result.success) {
+      throw new ValidationFailedError(this.operator, message, result.error);
+    }
+  }
+}
+
+class Equals extends BaseOperator {
+  constructor() {
+    super({
+      operator: 'EQUALS',
+      conditionValueSchema: PrimitiveSchema,
+      dataValueSchema: PrimitiveSchema,
+    });
+  }
+
+  eval: ConditionFn<primitive> = (dataValue: primitive, conditionValue: primitive): boolean => {
+    return dataValue === conditionValue;
+  };
+}
+
+class GreaterThan extends BaseOperator {
+  constructor() {
+    super({
+      operator: 'GT',
+      conditionValueSchema: PrimitiveSchema,
+      dataValueSchema: PrimitiveSchema,
+    });
+  }
+
+  eval: ConditionFn<primitive> = (dataValue: primitive, conditionValue: primitive): boolean => {
+    return dataValue > conditionValue;
+  };
+}
+
+class LessThan extends BaseOperator {
+  constructor() {
+    super({
+      operator: 'LT',
+      conditionValueSchema: PrimitiveSchema,
+      dataValueSchema: PrimitiveSchema,
+    });
+  }
+
+  eval: ConditionFn<primitive> = (dataValue: primitive, conditionValue: primitive): boolean => {
+    return dataValue < conditionValue;
+  };
+}
+
+class GreaterThanOrEqual extends BaseOperator {
+  equals: Equals;
+  greaterThan: GreaterThan;
+
+  constructor() {
+    super({
+      operator: 'GTE',
+      conditionValueSchema: PrimitiveSchema,
+      dataValueSchema: PrimitiveSchema,
+    });
+    this.equals = new Equals();
+    this.greaterThan = new GreaterThan();
+  }
+
+  eval: ConditionFn<primitive> = (dataValue: primitive, conditionValue: primitive): boolean => {
+    return (
+      this.equals.execute(dataValue, conditionValue) ||
+      this.greaterThan.execute(dataValue, conditionValue)
+    );
+  };
+}
+
+class LessThanOrEqual extends BaseOperator {
+  equals: Equals;
+  lessThan: LessThan;
+
+  constructor() {
+    super({
+      operator: 'LTE',
+      conditionValueSchema: PrimitiveSchema,
+      dataValueSchema: PrimitiveSchema,
+    });
+
+    this.equals = new Equals();
+    this.lessThan = new LessThan();
+  }
+
+  eval: ConditionFn<primitive> = (dataValue: primitive, conditionValue: primitive): boolean => {
+    return (
+      this.equals.execute(dataValue, conditionValue) ||
+      this.lessThan.execute(dataValue, conditionValue)
+    );
+  };
+}
+
+class Between extends BaseOperator<BetweenParams> {
+  gte: GreaterThanOrEqual;
+  lte: LessThanOrEqual;
+
+  constructor() {
+    super({
+      operator: 'BETWEEN',
+      conditionValueSchema: BetweenSchema,
+      dataValueSchema: PrimitiveSchema,
+    });
+    this.gte = new GreaterThanOrEqual();
+    this.lte = new LessThanOrEqual();
+  }
+
+  eval: ConditionFn<BetweenParams> = (
```

Comment:
eval is the functionality of one of the conversions of the values.
evaluate is a rule valuation function